Bottom: 6eacfb8f692575fb15f8c2006d59bbb8aec5580f
Top:    0368f77f26b4b41252408036e4de5b47bd3cd09a
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-25 14:57:05 -0500

Create new role nexus to configure Nexus3 service

---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index c306cadfa..88c2fbdcd 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -11,6 +11,9 @@ gitlab_secrets_db_key_base:          '{{ lookup("password", playbook_dir + "/cre
 mantisbt_db_password:                '{{ lookup("password", playbook_dir + "/credentials/mantis/db/user"                      , length=20) }}'
 mantisbt_db_root_password:           '{{ lookup("password", playbook_dir + "/credentials/mantis/db/root"                      , length=20) }}'
 mantisbt_email_password:             '{{ lookup("password", playbook_dir + "/credentials/mantis/email/mantis@novatech-llc.com", length=20) }}'
+nexus_admin_password:                '{{ lookup("password", playbook_dir + "/credentials/nexus/users/admin"                   , length=20) }}'
+nexus_buildbot_password:             '{{ lookup("password", playbook_dir + "/credentials/nexus/users/buildbot"                , length=20) }}'
+nexus_email_password:                '{{ lookup("password", playbook_dir + "/credentials/exim4/nexus@novatech-llc.com"        , length=20) }}'
 openldap_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/openldap/cn=admin,dc=novatech"       , length=20) }}'
 openldap_proxyagent_password:        '{{ lookup("password", playbook_dir + "/credentials/openldap/cn=proxyagent,dc=novatech"  , length=20) }}'
 svn_password:                        '{{ lookup("password", playbook_dir + "/credentials/svn/ansible"                         , length=20) }}'
diff --git a/ansible-playbook/group_vars/all/nexus3.yml b/ansible-playbook/group_vars/all/nexus3.yml
new file mode 100644
index 000000000..c3765ecf2
--- /dev/null
+++ b/ansible-playbook/group_vars/all/nexus3.yml
@@ -0,0 +1,153 @@
+nexus_version: 3.13.0
+nexus_networks:
+  - name: '{{ docker_network_frontend }}'
+    aliases:
+      - '{{ nexus_hostname }}'
+docker: true
+nexus_default_port: 80
+public_hostname: '{{ nexus_hostname }}'
+
+nexus_email_server_enabled: true
+nexus_email_server_host: "mail.novatech-llc.com"
+nexus_email_server_port: 587
+nexus_email_server_username: "wiki"
+nexus_email_server_password: "{{ nexus_email_password }}"
+nexus_email_from_address: "nexus@novatech-llc.com"
+nexus_email_subject_prefix: "[NEXUS]"
+nexus_email_tls_enabled: true
+nexus_email_tls_required: true
+
+nexus_ldap_realm: true
+ldap_connections:
+  - ldap_name: 'NovaTech LDAP' # used as a key to update the ldap config
+    ldap_protocol: 'ldap' # ldap or ldaps
+    ldap_hostname: 'ldap.novatech-llc.com'
+    ldap_port: 389
+    ldap_auth: 'simple'
+    ldap_auth_username: 'cn=proxyagent,dc=novatech'
+    ldap_auth_password: '{{ openldap_proxyagent_password }}'
+    ldap_search_base: 'dc=novatech'
+    ldap_user_base_dn: 'ou=user'
+    #ldap_user_filter: '(cn=*)' # (optional)
+    ldap_user_object_class: 'inetOrgPerson'
+    ldap_user_id_attribute: 'uid'
+    ldap_user_real_name_attribute: 'cn'
+    ldap_user_email_attribute: 'mail'
+    ldap_group_base_dn: 'ou=group'
+    ldap_group_object_class: 'posixGroup'
+    ldap_group_id_attribute: 'gidNumber'
+    ldap_group_member_attribute: 'memberUid'
+    ldap_group_member_format: '${username}'
+
+nexus_privileges:
+  - name: all-repos-read  # used as key to update a privilege
+    description: 'Read & Browse access to all repos'
+    repository: '*'
+    actions:
+      - read
+      - browse
+
+nexus_roles:
+  - id: Developers  # can map to a LDAP group id, also used as a key to update a role
+    name: developers
+    description: All developers
+    privileges:
+      - nx-search-read
+      - all-repos-read
+    roles: []
+
+nexus_local_users:
+  - username: buildbot # used as key to update
+    first_name: NovaTech
+    last_name: CI
+    email: buildbot@novatech-llc.com
+    password: "{{ nexus_buildbot_password }}"
+    roles:
+      - Developers # role ID here
+
+nexus_delete_default_blobstore: false
+nexus_blob_split: false     # True - blobstores per format
+
+nexus_blobstores:
+- name: permanent
+  path: '{{ nexus_data_dir }}/blobs/permanent'
+
+nexus_delete_default_repos: true
+
+nexus_config_maven: false
+
+nexus_config_pypi: true
+nexus_repos_pypi_hosted:
+  - name: pypi-hosted
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+nexus_repos_pypi_proxy:
+  - name: pypi-proxy
+    remote_url: 'https://pypi.python.org/'
+nexus_repos_pypi_group:
+  - name: pypi
+    member_repos:
+      - pypi-hosted
+      - pypi-proxy
+
+nexus_config_docker: true
+nexus_repos_docker_hosted:
+  - name: docker-hosted
+    http_port: "{{ nexus_docker_hosted_port }}"
+    v1_enabled: True
+nexus_repos_docker_proxy:
+  - name: docker-proxy
+    http_port: "{{ nexus_docker_proxy_port }}"
+    v1_enabled: True
+    index_type: "HUB"
+    proxy_url: "https://registry-1.docker.io"
+    use_nexus_certificates_to_access_index: false
+nexus_repos_docker_group:
+  - name: docker
+    http_port: "{{ nexus_docker_group_port }}"
+    v1_enabled: True
+    member_repos:
+      - docker-hosted
+      - docker-proxy
+
+nexus_config_raw: true
+nexus_repos_raw_hosted:
+  # Global download storage
+  - name: download
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+    blob_store: permanent
+  - name: ntel-release
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+    blob_store: permanent
+  - name: ntel-sstate
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+  - name: ntel-unofficial
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+  - name: ntel-sstate
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+    blob_store: permanent
+  - name: ptxdist-release
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+    blob_store: permanent
+  - name: ptxdist-unofficial
+    strict_content_validation: false
+    write_policy: allow  # one of "allow", "allow_once" or "deny"
+
+nexus_repos_raw_proxy:
+  - name: ubuntu-archive
+    remote_url: 'http://archive.ubuntu.com/ubuntu/'
+    strict_content_validation: false
+
+nexus_repos_raw_group: []
+
+nexus_config_rubygems: false
+nexus_config_bower: false
+nexus_config_npm: false
+nexus_config_nuget: false
+nexus_config_gitlfs: false
+nexus_config_yum: false
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index f908c8f03..e94bb8ac4 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -47,3 +47,8 @@ phpldapadmin_port_args:
 - "{{ phpldapadmin_ip_addr }}:443:443"
 
 phpmyadmin_hostname: phpmyadmin.novatech-llc.com
+
+nexus_hostname: nexus.novatech-llc.com
+nexus_port_args:
+- "{{ nexus_ip_addr }}:80:8081"
+- "{{ nexus_ip_addr }}:443:8443"
diff --git a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
index c63231984..2d8c4735f 100644
--- a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
@@ -15,3 +15,4 @@ exim4_ip_addr:        172.16.0.107
 openldap_ip_addr:     172.16.0.108
 phpldapadmin_ip_addr: 172.16.0.108
 phpmyadmin_ip_addr:   172.16.0.109
+nexus_ip_addr:        172.16.0.111
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
index b3e366ced..8d8fabe4f 100644
--- a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
@@ -15,3 +15,4 @@ exim4_ip_addr:        172.16.103.107
 openldap_ip_addr:     172.16.103.108
 phpldapadmin_ip_addr: 172.16.103.108
 phpmyadmin_ip_addr:   172.16.103.109
+nexus_ip_addr:        172.16.103.111
